File: k12ms26/greedy/회의실배정.js
//BOJ 1931 회의실배정
const solution = (input) => {
    const N = input.shift();
    const arr = input.map((v) => v.split(' '));
    arr.sort((a, b) => a[1]-b[1] || a[0]-b[0]);
    let end_time = 0, cnt = 0, e = [];
    for(let [start, end] of arr) {
        if(+start >= end_time) {
            cnt++;
            e.push([start, end])
            end_time = end;
        }
    }
    console.log(cnt);
}

const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

const input = [];
rl.on("line", (line) => {
    input.push(line.trim());
}).on("close", () => {
    solution(input);
    rl.close();
    process.exit();
});

File: k12ms26/greedy/회의실배정.py
#회의실 배정
import sys

N = int(sys.stdin.readline())
time = []
end_time = 0; cnt = 0
for i in range(N) :
    a, b = map(int,sys.stdin.readline().split())
    time.append([a,b])

time = sorted(time, key=lambda a : a[0])
time = sorted(time, key=lambda a : a[1])
print(time)
for i, j in time :
    if i >= end_time :
        cnt += 1
        end_time = j

print(cnt)